elip-0100.mediawiki

```diff
@@ -64,23 +64,41 @@ This specification relies on definition of roles in the main PSET standard. The
 
 ===Updater===
 
-Besides the responsibilities defined in PSET specification, Updater should add asset metadata for each asset referenced in any of inputs and outputs. This information is typically queried from the global asset registry, but cached or pre-defined information could be used as well. For each asset Updater adds a separate <tt>PSBT_ELEMENTS_HWW_GLOBAL_ASSET_METADATA</tt> field using asset tag as keydata. It is desirable however not critical for the Updater to verify asset metadata if it is received from an untrusted source. This verification is obligatory for the Signer role.
+Besides the responsibilities defined in PSET specification, Updater should add asset metadata for each asset referenced in any of inputs and outputs. This information is typically queried from the global asset registry, but cached or pre-defined information could be used as well. For each asset, the Updater adds a separate <tt>PSBT_ELEMENTS_HWW_GLOBAL_ASSET_METADATA</tt> field using asset tag as keydata. It is desirable however not critical for the Updater to verify asset metadata if it is received from an untrusted source. This verification is obligatory for the Signer role.
```

Comment:
> referenced in any of inputs and outputs

I think this should be `referenced in any inputs`, unless metadata is meant to be provided for issuance txs, in which case it should be `referenced in any inputs or issuance outputs`.

Reply:
For Elements transactions, an asset tag or commitment is specified explicitly for every input as well as for every output. This is needed because a transaction may spend multiple UTXOs having different assets, distributing them to different addresses as swap transactions do you mentioned earlier. The simpler example is an ordinary USDt transaction, requiring an additional L-BTC input to pay the fee and usually an additional L-BTC output for the change.

Reply:
I'm aware that outputs have asset tags, my point is that only issuance txs can create new ones. Therefore providing the asset tags for the inputs is sufficient.

That is, unless you want to provide the metadata for actual issuance assets too, in which case only the output(s) that create new assets need it, not all outputs.